### DISCUSSION TITLE
Display more information for the "Fix validation errors on the related asset." message

### DISCUSSION THREAD

**Discussion by MoritzLost** [craftcms/cms#9960] [Ideas] [answered]
Craft provides a great feature to prevent an element from saving if related elements have validation errors. However, if there are validation errors in a related element, the error message is pretty technical and doesn't convey enough information to be useful to non-technical folk: "Fix validation errors on the related asset."

For example, consider this scenario: We have an asset volume for images with an required field for the image's alt text (used as the `alt` attribute). So when an editor uploads an image directly through an entry field and tries to save the entry, they get the error message above. The problem is that the specific validation errors (in this case, that a required field is empty) are not displayed, so they don't know what to do. Instead, they assume it's a bug and contact the developers to fix it. This is corroborated by the fact that there's no indication that the image can be edited with a double-click.

We try to use the instructions field to make this more clear, but people don't read …

My proposed solution would be to display the specific validation errors of the related asset inline below the generic message above. That would indicate to editors that it's not a bug, but a content-related error that they can fix.

### Current message

> Fix validation errors on the related asset.

### Proposed change

> Fix the following validation errors on the related asset:
> - Alt-Text cannot be blank.

**Comment by MoritzLost** [comment]
@brandonkelly @i-just Any update on this? I need more than two hands to count the number of clients who did not get it and required an explanation from us. The current error message is vague, technical, and badly worded.

> Validation errors found in {field} asset; please fix them.

This is not a good author experience, and also not accessible. We really need better AX here.

## The message should be clear and actionable

For example:

> The selected asset contains validation errors that prevent this entry from being saved. Open the selected asset to fix them.

Might need some adjustments for plurals, different element types etc, but you get the idea. This makes it clear that this is not a misconfiguration that you need to call your developer about, but a content issue that you can fix yourself.

## What's wrong with the related asset?

The error should show a list or summary of validation errors on the related asset(s).

Showing a list would be useful:

> The selected asset contains validation errors that prevent this entry from being saved. Open the selected asset to fix the following errors:
> - {field} cannot be blank.

As an alternative, or maybe if there are a lot of errors, or errors from multiple related assets, it could show a summary instead:

> The selected asset contains validation errors that prevent this entry from being saved. Open the selected asset to fix the following errors:
> - 2 required fields are missing

Either of those would make it immediately obvious what the problem is, and how it can be addressed.

---

I realize the second suggestion might be tough to implement. But at least the first suggestion is immediately actionable and would be a great improvement.

**Reply by brandonkelly** [reply, marked as answer]
Agree, that’s a nice improvement.

Went with something similar for Craft 4.16 and 5.8: (adb6229fcdf1c9b2accc88caf223bc75edc64278)

![An Assets field with two selected assets, showing the validation error “The selected assets contain validation errors, preventing this entry from being saved. Edit the assets to fix them.”](https://github.com/user-attachments/assets/b29dc1fc-3c53-42c5-8922-e2aa820b88fa)

Note that we can’t show a nested bullet list, without introducing a breaking change that could also impact front-end templates. Guessing this will give authors enough direction to go on, though.

**Reply by MoritzLost** [reply]
@brandonkelly That look great, thanks, Brandon! A summary or nested error list would be nice, but the improved error message will already be a huge improvement, and will probably suffice for most authors 👍

**Reply by brandonkelly** [reply]
Craft 4.16 and 5.8 are out with that change.
